numpyro/util.py

```diff
@@ -309,3 +311,254 @@ def soft_vmap(fn, xs, batch_ndims=1, chunk_size=None):
+        'plate_groups': plate_groups,
+        'plate_data': plate_data,
+        'node_data': node_data,
+        'edge_list': edge_list,
+    }
+
+
+def render_graph(graph_specification, render_distributions=False):
+    """
+    Create a graphviz object given a graph specification.
+
+    :param bool render_distributions: Show distribution of each RV in plot.
+    """
+    try:
+        import graphviz  # noqa: F401
+    except ImportError as e:
+        raise ImportError(
+            'Looks like you want to use graphviz to render your model. '
+            'You need to install `graphviz` to be able to use this feature. '
+            'It can be installed with `pip install graphviz`.'
+        ) from e
+
+    plate_groups = graph_specification['plate_groups']
+    plate_data = graph_specification['plate_data']
+    node_data = graph_specification['node_data']
+    edge_list = graph_specification['edge_list']
+
+    graph = graphviz.Digraph()
+
+    # add plates
+    plate_graph_dict = {
+        plate: graphviz.Digraph(name=f'cluster_{plate}')
+        for plate in plate_groups
+        if plate is not None
+    }
+    for plate, plate_graph in plate_graph_dict.items():
+        plate_graph.attr(label=plate.split('__CLONE')[0], labeljust='r', labelloc='b')
+
+    plate_graph_dict[None] = graph
+
+    # add nodes
+    for plate, rv_list in plate_groups.items():
+        cur_graph = plate_graph_dict[plate]
+
+        for rv in rv_list:
+            color = 'grey' if node_data[rv]['is_observed'] else 'white'
+            cur_graph.node(
+                rv, label=rv, shape='ellipse', style='filled', fillcolor=color
+            )
+
+    # add leaf nodes first
+    while len(plate_data) >= 1:
+        for plate, data in plate_data.items():
+            parent_plate = data['parent']
+            is_leaf = True
+
+            for plate2, data2 in plate_data.items():
+                if plate == data2['parent']:
+                    is_leaf = False
+                    break
+
+            if is_leaf:
+                plate_graph_dict[parent_plate].subgraph(plate_graph_dict[plate])
+                plate_data.pop(plate)
+                break
+
+    # add edges
+    for source, target in edge_list:
+        graph.edge(source, target)
+
+    # render distributions if requested
+    if render_distributions:
+        dist_label = ''
+        for rv, data in node_data.items():
+            rv_dist = data['distribution']
+            dist_label += f'{rv} ~ {rv_dist}\l'
+
+        graph.node(
+            'distribution_description_node', label=dist_label, shape='plaintext'
+        )
+
+    # return whole graph
+    return graph
+
+
+def render_model(
+    model,
+    model_args=None,
+    model_kwargs=None,
+    filename=None,
+    render_distributions=False,
+    num_tries=10,
+):
+    """
+    Wrap all functions needed to automatically render a model.
+
+    :param model: Model to render.
+    :param str filename: File to save rendered model in.
```

Comment:
The format can already be influenced via the filename (e.g., `model.pdf` vs `model.png`).
Would it be enough to simply add a note for this in the documentation?

Reply:
Oh, great! Then I think we don't need format argument. In the notebook, could you display `png` format so that the diff is small between different commits?